server/channels/api4/websocket_test.go

```diff
@@ -536,3 +537,147 @@ func TestValidateDisconnectErrCode(t *testing.T) {
 		})
 	}
 }
+
+// Helper function to enable MFA enforcement in config
+func enableMFAEnforcement(th *TestHelper) {
+	th.App.UpdateConfig(func(cfg *model.Config) {
+		*cfg.ServiceSettings.EnableMultifactorAuthentication = true
+		*cfg.ServiceSettings.EnforceMultifactorAuthentication = true
+	})
+}
+
+// Helper function to set up MFA for a user
+func setupUserWithMFA(t *testing.T, th *TestHelper, user *model.User) string {
+	// Setup MFA properly - following authentication_test.go pattern
+	secret, appErr := th.App.GenerateMfaSecret(user.Id)
+	require.Nil(t, appErr)
+	err := th.Server.Store().User().UpdateMfaActive(user.Id, true)
+	require.NoError(t, err)
+	err = th.Server.Store().User().UpdateMfaSecret(user.Id, secret.Secret)
+	require.NoError(t, err)
+	return secret.Secret
+}
+
+func TestWebSocketMFAEnforcement(t *testing.T) {
+	mainHelper.Parallel(t)
+
+	t.Run("WebSocket works when MFA enforcement is disabled", func(t *testing.T) {
+		th := Setup(t).InitBasic()
+		defer th.TearDown()
+
+		// MFA enforcement disabled - should work normally
+		webSocketClient := th.CreateConnectedWebSocketClient(t)
+		defer webSocketClient.Close()
+
+		webSocketClient.GetStatuses()
+
+		select {
+		case resp := <-webSocketClient.ResponseChannel:
+			require.Nil(t, resp.Error, "WebSocket should work when MFA enforcement is disabled")
+			require.Equal(t, resp.Status, model.StatusOk)
+		case <-time.After(3 * time.Second):
+			require.Fail(t, "Expected WebSocket response but got timeout")
+		}
+	})
+
+	t.Run("WebSocket blocked when MFA required but user has no MFA", func(t *testing.T) {
+		th := SetupEnterprise(t).InitBasic()
+		defer th.TearDown()
+
+		// Enable MFA enforcement in config
+		enableMFAEnforcement(th)
+		// Defer the teardown to reset the config after the test
+		defer func() {
+			th.App.UpdateConfig(func(cfg *model.Config) {
+				*cfg.ServiceSettings.EnforceMultifactorAuthentication = false
+			})
+		}()
+
+		// Create user without MFA using existing basic user to avoid license timing issues
+		user := th.BasicUser
+
+		// Login user (this should work for initial authentication)
+		client := th.CreateClient()
+		_, _, err := client.Login(context.Background(), user.Email, "Pa$$word11")
+		require.NoError(t, err)
+
+		// Create WebSocket client without expecting hello event (MFA enforcement prevents it)
```

Comment:
Pursuant to my comment above, it looks like we went ahead and implemented this? What are the consequences to mobile and webapp flows around enabling MFA and re-establishing the websocket connection? Should we just revert this requirement?

Reply:
Should this test have changed after the revert, @catalintomai? (Surprised it didn't fail before and/or had to be changed now.)

Reply:
@lieut-data , if I understand correctly the q, for this test the user has both a valid session and has completed MFA authentication. So, "before", when we were using "IsBasicAuthenticated() && IsMFAAuthenticated()" , we had true and "after" revert, when we use IsBasicAuthenticated only, we have true. We'd need to add a different test that captures "user has NOT completed MFA (but MFA is required)". In which case, the "before" should be failing and the "after" passes.